[docs] Add how to use existing bitcoin data_dir for the btcpay docker

BobleChinois opened the pull request:
I suggested in an [issue on btcpayserver-docker](https://github.com/btcpayserver/btcpayserver-docker/issues/34) that we find a way to use an already synced blockchain with BTCPay when possible.

The point is to allow people that already run a node to avoid a long and costly resync and leverage the work already done for their BTCPay full node.

I made a few modifications to `docker-compose.generated.yml`, and made BTCPay mount my existing `.bitcoin` directory instead of the docker volume `bitcoin_datadir`.

It doesn't seem to represent any risk, I was able to reuse the same directory with plain `bitcoind` thereafter, and then again with BTCPay without issue.

I propose this very easy "how to" as an addition to the FAQ, waiting maybe for a more durable solution one day.

NicolasDorier commented:
this solution is not very good, using a softlink instead would not require renaming (https://stackoverflow.com/questions/9587445/how-to-create-a-link-to-a-directory)

BobleChinois commented:
Ok, just to be sure I got it right, what you mean is that I should create a symlink from docker's volume to my existing .bitcoin, so that I don't need to modify the .yml file, is that correct ?

BobleChinois commented:
I tried to do that, tracked the location of `generated_bitcoin_datadir` on my host with `docker volume inspect`, removed the automatically generated volume and created a symlink to my `.bitcoin` directory with `ln -s /home/sosthene/.bitcoin2 /var/lib/docker/volumes/generated_bitcoin_datadir`

Next time I started, it created a `_data` dir inside my `.bitcoin2`, and then all the directories (blocks, chainstate...) inside the `_data`. 

So I stopped it, rm all the content of the `_data` dir, and moved all of `.bitcoin2` inside `_data`.

I started again btcpay, and it seems to work this way.

I'm doing a few more test to be sure.

BobleChinois commented:
Yes, it should obviously work better. I'll do it with another machine to see if it still works. 

I have a question though, how is doing things this way better than modifying the .yml file ? Just to understand :smile:

BobleChinois commented:
I tested it with another machine and confirm that using symlink works for me. 

I propose another "how to" for the FAQ, reproducing the steps I have done myself.

NicolasDorier commented:
awesome
